content/knowledge_base/flare/consensus/avalanche/specification.md

```diff
@@ -0,0 +1,210 @@
+Oppositely — the term "progeny" means all existing (and potentially existing) children transactions and their children.
+
+The main purpose of a consensus protocol is to avoid the inclusion of conflicting transactions into the ledger.
+Even though the characteristics of a conflicting transaction are defined on an application level (ex. transactions that spend same UTXO), the notion of conflict can be abstracted in order to define the conflicting set.
+In Avalanche, every transaction belongs to a conflict set.
+This set consists of multiple transactions that are invalid towards each other or it is a singleton set - then it contains a single transaction (in the case of a cirtuous transaction).
+Since reaching a consensus means to avoid any conflicts between transactions, only one transaction from each conflict set can be included in the approved path of transactions.
+
+Avalanche instantiates a Snowball instance for each conflict set.
+Avalanche treats the concept of repeated queries and multiple counters from Snowball taking advantage of DAG structure.
+Specifically, when a transaction `T` is queried, all of its ancestors reachable through the DAG edges from it are implicitly part of the query.
+Consequently, nodes would respond positively to a query for transaction `T` only if `T` and all of its ancestors are the preferred option in their respective conflict sets.
+If more than a given threshold of responders vote positively, the transactions gets a [chit](../../../glossary.md#Chit).
+After that nodes compute their level of confidence as the sum of transaction chit and the chits its progeny, meaning they query the transaction just once and rely on new vertices and possible chits added to the progeny to build up their confidence.
+
+Described ties are broken in case of preference for first-seen transactions. TODO: explain this
+Chits can also be decoupled from the DAG structure, making the protocol immune to attacks where the attacker generates large padded subgraphs.
+
+## Consensus Specification
+
+Each node `u` keeps track of all transactions it has learned about in set Tu. The set Tu is [partitioned](../../../glossary.md#Partition-of-a-Set) into mutually exclusive conflict sets - PT (partition of 'T'), and PT ∈ Tu.
+Since conflicts are [transitive](../../../glossary.md#Transitive-Relation), if PTi and PTj are conflicting they belong to the same conflict set of transactions — PTi = PTj.
+Conflicting transation have the equivalence relation because they are equivocations spending the same UTXO.
+
+The parent connection to T' from T is expressed as `T' <— T`.
+The relation `T' *<— T` is its reflexive [transitive closure](../../../glossary.md#Transitive-Closuse-of-a-Graph) meaning that there is path from T to T'.
+If T' <— T, then every node in the system that has T will also have T' and will know about the same relation `T' <— T`.
+If such a relation between these transactions do not exist, then no node will end up with T' <— T.
```

Comment:
```suggestion
If such a relation between these transactions do not exist, then no node ends up with `T' <— T`.
```

Reply:
This is corrected https://github.com/optakt/optakt.github.io/pull/13/commits/e0e27e6be8c0aef4153af5e2add9722380c23a59#diff-2d4f16e70781857aec6689a748ad155e45611b81f184aee076bb139890410b98R56
Thanks!